src/hotspot/share/oops/oop.hpp

```diff
@@ -108,6 +121,17 @@ class oopDesc {
   // to be able to figure out the size of an object knowing its klass.
   inline size_t size_given_klass(Klass* klass);
 
+  // The following set of methods is used to access the mark-word and related
+  // properties when the object may be forwarded. Be careful where and when
+  // using this method. It assumes that the forwardee is installed in
+  // the header as a plain pointer (or self-forwarded). In particular,
+  // those methods can not deal with the sliding-forwarding that is used
+  // in Serial, G1 and Shenandoah full-GCs.
+  inline markWord forward_safe_mark() const;
```

Comment:
`forward_safe_mark` seems to be only used in `forward_safe_klass`, can be inlined/simplified there?